base/steer/FairRun.h

```diff
@@ -222,6 +228,10 @@ class FairRun : public TNamed
 
     Bool_t                   fMarkFill; //!
 
+    FairAlignmentHandler fAlignmentHandler;
+
+    void AlignGeometry() const;
```

Comment:
Let's just make this new member function `public`, so we do not need the explicit tight coupling of `FairRun` and `FairMCApplication` introduced in line `base/steer/FairRun.h#34`.

Reply:
Hmm, I wanted to avoid users from calling `AlignGeometry()` of FairRun. What is so bad about making FairMCApplication a friend class?

Reply:
I just don't see the tight relationship of `FairRun` and `FairMCApplication`. Maybe we could move the friend declaration to `FairRunSim`?

Reply:
I would prefer to leave the friend declaration in the FairRun. For most of the simulation macros we use FairRunSim, that's true, but we also need the alignment functionality in the Digi macros, which use a FairRunAna, not a FairRunSim. Since both of these inherit from FairRun, is seems best to include the functionality in FairRun.

The reason we need it in the Digi macros is that the digitizer must know the sensors are in the wrong place, otherwise it will just take the mc points that come from the misaligned geometry and calculate digi hits that are outside the pixel detector (and will subsequently have out of bounds row and col values).

Reply:
In this case, I would recommend to just remove it completely and declare the member function public.

But in the end, I am not the decider in this matter ;)

Reply:
I don't like the friend class that much either, as it suggests the code as a whole maybe bad design. But we could not come up with something better, unless you redesign large portions of FairRoot or ROOT.
Making the function public is just moving the problem from one part to another. I mean we can put a wrapper around the AlignGeometry() to reject calls that would lead to problems (which the user could make). But thats not super pretty either.
We can just explain our idea or way of thinking. You guys can discuss internally and let us know what you would like most. Did you read my general comment about the how to read the alignment parameters?